core/new-gui/src/app/dashboard/service/workflow-snapshot/workflow-snapshot.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from "@angular/core";
+import { Observable } from "rxjs";
+import { AppSettings } from "../../../common/app-setting";
+import { HttpClient } from "@angular/common/http";
+import html2canvas from "html2canvas";
+import { WorkflowSnapshotEntry } from "../../type/workflow-snapshot-entry";
+
+export const WORKFLOW_SNAPSHOT_API_BASE_URL = `${AppSettings.getApiEndpoint()}/snapshot`;
+export const WORKFLOW_SNAPSHOT_UPLOAD_URL = `${WORKFLOW_SNAPSHOT_API_BASE_URL}/upload`;
+
+@Injectable({
+  providedIn: "root",
+})
+export class WorkflowSnapshotService {
+  constructor(private http: HttpClient) {}
+
+  /**
+   * create canvas for snapshot
+   */
+  public createSnapShotCanvas(): Promise<HTMLCanvasElement> {
+    let doc = document.getElementById("texera-workflow-editor") || document.body;
+    const { height, width } = doc.getBoundingClientRect();
+    return html2canvas(doc, {
+      allowTaint: true,
+      useCORS: true,
+      backgroundColor: "transparent",
+      height: height * 0.6,
```

Comment:
can you please avoid scaling the image here by hardcoding the numbers ).6, 0.2, 0.7 etc and give variables in the file to specify the scale

Reply:
done